Public/Src/Cache/ContentStore/Distributed/NuCache/RocksDbContentLocationDatabase.cs

```diff
@@ -507,6 +529,197 @@ private byte[] GetKey(ShortHash hash)
             return hash.ToByteArray();
         }
 
+        // TODO(jubayard): garbage collection / removal in general
+
+        /// <inheritdoc />
+        public override GetContentHashListResult GetContentHashList(OperationContext context, StrongFingerprint strongFingerprint)
+        {
+            return context.PerformOperation(
+                Tracer,
+                () =>
+                {
+                    var key = GetMetadataKey(strongFingerprint);
+                    ContentHashListWithDeterminism? result = null;
+                    var status = _keyValueStore.Use(
+                        store =>
+                        {
+                            if (store.TryGetValue(key, out var data, nameof(Columns.Metadata)))
+                            {
+                                result = DeserializeContentHashList(data);
+                                // TODO(jubayard): since we are inside the ContentLocationDatabase, we can validate that all
+                                // hashes exist. Moreover, we can prune content.
+                            }
+                        });
+
+                    if (!status.Succeeded)
+                    {
+                        return new GetContentHashListResult(status.Failure.CreateException());
+                    }
+
+                    if (result == null)

```

Comment:
Minor: Small optimization for null-checking: it's better to write (result is null) than (result == null), particularly on a type that overrides equals. This comment applies to a lot of places.